### PR TITLE
[WIP] Experimental Graph DSL

### PR DESCRIPTION
**Pull request by peterbourgon**
Pending merge of #399

[Review on Reviewable](https://reviewable.io/reviews/weaveworks/scope/407)

### CODE REVIEW

File: render/theinternet.go [hunk 1]
```diff
@@ -13,27 +13,33 @@ import (
...
-			if err != nil {
-				continue
-			}
-			_, ok := networks[ipNet.String()]
-			if !ok {
-				result = append(result, ipNet)
-				networks[ipNet.String()] = struct{}{}
-			}
-		}
+		result = append(result, ParseNetworks(val)...)
```
Comment (tomwilkie) [inline render/theinternet.go]: Won't this result in multiple entries in result if the hosts are on the same networks?

File: experimental/dsl/lexer.go [hunk 2]
```diff
@@ -0,0 +1,326 @@
+package main
```
Comment (tomwilkie) [inline experimental/dsl/lexer.go]: Shouldn't we be using yacc & lex for this?

File: render/theinternet.go [hunk 3]
```diff
@@ -13,27 +13,33 @@ import (
 // used to determine which nodes in the report are "remote", i.e. outside of
 // our infrastructure.
 func LocalNetworks(r report.Report) report.Networks {
-	var (
-		result   = report.Networks{}
-		networks = map[string]struct{}{}
-	)
-
+	result := report.Networks{}
```
Comment (tomwilkie) [inline render/theinternet.go]: These changes seem unrelated to this PR?

File: render/theinternet.go (hunk 1 above)
Comment (peterbourgon) [inline render/theinternet.go]: Yep, good spot. Thanks.

File: render/theinternet.go (hunk 3 above)
Comment (peterbourgon) [inline render/theinternet.go]: I needed ParseNetworks broken out, this was just a change for that.

File: experimental/dsl/lexer.go (hunk 2 above)
Comment (peterbourgon) [inline experimental/dsl/lexer.go]: Nah :)
